qa/rpc-tests/mvf-bu-retarget.py

```diff
@@ -41,98 +129,106 @@ def is_fork_triggered_on_node(self, node=0):
         return (len(hf_active) > 0 and len(fork_actions_performed) == 1)
 
     def run_test(self):
-        # check that fork does not triggered before the height
+        # check that fork does not trigger before the forkheight
         print "Generating %s pre-fork blocks" % (FORK_BLOCK - 1)
-
+        #block0 already exists
         for n in range(FORK_BLOCK - 1):
             # Change block times so that difficulty develops
             best_block = self.nodes[0].getblock(self.nodes[0].getbestblockhash(), True)
-            #print "%s %s" %(
-                #best_block['height'],
-                #time.strftime("%Y-%m-%d %H:%M",time.gmtime(best_block['time'])))
-            self.nodes[0].setmocktime(best_block['time'] + 600)
+            self.nodes[0].setmocktime(best_block['time'] + PREFORK_BLOCKTIME)
             self.nodes[0].generate(1)
 
         # Read difficulty before the fork
         best_block = self.nodes[0].getblock(self.nodes[0].getbestblockhash(), True)
-        print "Pre-fork difficulty: %s %s " % (round(best_block['difficulty'],8), best_block['bits'])
-        best_diff_expected = best_block['difficulty'] / 10
+        print "Pre-fork difficulty: %.10f %s " % (best_block['difficulty'], best_block['bits'])
+        nBits = CalculateMVFResetWorkRequired(best_block['bits'])
+        reset_bits = int2hex(nBits)
+        reset_diff_expected = bits2difficulty(nBits)
+        assert_greater_than(reset_diff_expected, 0)
 
         # Test fork did not trigger prematurely
         assert_equal(False, self.is_fork_triggered_on_node(0))
         print "Fork did not trigger prematurely"
 
         # Generate fork block
         best_block = self.nodes[0].getblock(self.nodes[0].getbestblockhash(), True)
-        self.nodes[0].setmocktime(best_block['time'] + 600)
+        self.nodes[0].setmocktime(best_block['time'] + STANDARD_BLOCKTIME)
         self.nodes[0].generate(1)
         assert_equal(True,   self.is_fork_triggered_on_node(0))
-        print "Fork triggered successfully (block height %s)" % FORK_BLOCK
+
+        best_block = self.nodes[0].getblock(self.nodes[0].getbestblockhash(), True)
+        print "Fork triggered successfully (block height %s)" % best_block['height']
+
+        # Test fork difficulty reset
+        #print "expected = %.10f actual = %.10f" % (reset_diff_expected,best_block['difficulty'])
+        assert_equal(best_block['bits'],reset_bits)
+        #assert_equal(best_block['bits'], "207eeeee") # fixed reset
+        print "Post-fork difficulty reset success: %.10f %s " % (best_block['difficulty'], best_block['bits'])
 
         # use to track how many times the same bits are used in a row
         prev_block = 0
         diffadjinterval = 0
-        next_block_time = 0
+        next_block_time = 300
```

Comment:
It can start from anywhere as it goes through a phase of adding 50 to the blocktime each cycle But if its too low for too many blocks then its likely to cause a raise in difficulty that leads to a timeout during the test. I raised it to a level that should always avoid timeouts.